### DISCUSSION TITLE
exp(shap.expected_value) very different than mean(exp(predictions)) where prediction is in log scale

### DISCUSSION THREAD

**Discussion by ananya-8451** [shap/shap#3011] [Q&A]
I have been working on a model where we have log transformed the target variable. 
Then we have used shap so our shap outputs are on log scale. Now when I calculate the 

```
# exp(mean of training predictions in log scale)
print(np.exp(model.predict(train_df[feature_list]).mean())) 
output = 56.40563
```

```
# mean(exp of training predictions in log scale)
print((np.exp(model.predict(train_df[feature_list]))).mean())
output = 240.56863
```

Now, I am not sure which one is the correct expected value to use? Do we need a bias correction here to represent the true base value? But, in that case do we violate the additivity principle?
My ultimate goal here is to get the shap ouputs to the original scale why I am trying to figure out the correct base value to use.

**Comment by thatlittleboy** [comment]
hi @ananya-8451 SHAP explains outputs in the scale of the original model output (i.e., ur log-transformed y) by default. I don't think any form of non-linear transformation to the shap values will preserve the additive property.

Consider the reply by Scott in this issue, https://github.com/slundberg/shap/issues/29#issuecomment-371877433 (though in that issue, it's more about transforming log-odds in a binary classification problem back to probabilities via the logistic function, but the general principle holds).

If you transform the base value (the first option you gave), you'll have to transform the shap values also, but that's an approximation, not exact.
Btw, the base value should be `train_y.mean()`, no? (before you consider any transformations).
